IPython/html/services/contents/filemanager.py

```diff
@@ -309,6 +315,24 @@ def _notebook_model(self, path, content=True):
             model['format'] = 'json'
             self.validate_notebook_model(model)
         return model
+        
+    def _read_notebook(self, os_path, as_version=4):
+        """Read a notebook from an os path."""
+        with self.open(os_path, 'r', encoding='utf-8') as f:
```

Comment:
Same thought as below, I think this can be accomplished with subclass implementation that looks like:
```
def _read_notebook(self, os_path, as_version=4):

    file_ext = _file_extension(os_path)
    if file_ext == '.md':
        do_special_markdown_stuff()
    else:
        return super(FileContentsManager, self)._read_notebook(os_path)
```